wp-admin/js/widgets/media-widgets.js

```diff
@@ -370,6 +367,7 @@ wp.mediaWidgets = ( function( $ ) {
 
 			if ( ! control.templateRendered ) {
 				control.$el.html( control.template()( control.model.attributes ) );
+				control.renderPreview(); // Hereafter it will re-render when control.selectedAttachment changes.
```

Comment:
@westonruter Brilliant, thank you! I was struggling to find where this re-rendering was failing, this makes way more sense.